src/cmap/test/mod.rs

```diff
@@ -250,17 +283,19 @@ impl Operation {
                     });
             }
             Operation::Clear => {
-                let mut subscriber = state.handler.subscribe();
-
-                if let Some(pool) = state.pool.write().await.deref() {
+                if let Some(pool) = state.pool.read().await.as_ref() {
+                    let mut subscriber = pool.subscribe_to_generation_updates();
                     pool.clear();
-                    // wait for event to be emitted to ensure drop has completed.
+
                     subscriber
-                        .wait_for_event(EVENT_TIMEOUT, |e| {
-                            matches!(e, Event::ConnectionPoolCleared(_))
-                        })
+                        .wait_for_generation_change(EVENT_TIMEOUT)
                         .await
-                        .expect("did not receive ConnectionPoolCleared event after clearing pool");
+                        .expect("generation did not change after clearing pool");
```

Comment:
Because clearing a cleared pool doesn't necessarily emit an event but always bumps the generation, we use the generation subscriber here to determine if the call occurred or not.